string.go

```diff
@@ -30,40 +28,45 @@ import (
 	"unicode/utf8"
 )
 
-// AESEncrypt encrypts text and given key with AES.
-func AESEncrypt(key, text []byte) ([]byte, error) {
+// AESEncrypt encrypts plaintext with the given key and nonce using AES in GCM mode.
+func AESEncrypt(key, nonce, plaintext []byte) ([]byte, error) {
```

Comment:
Quoting [NIST Special Publication 800-38D](http://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-38d.pdf), 8.3 "Constraints on the Number of Invocations":
> The total number of invocations of the authenticated encryption function shall not exceed 2^32, including all IV lengths and all instances of the authenticated encryption function with the given key.